internal/functions/metricsapi/prepare-new-version.go

```diff
@@ -0,0 +1,228 @@
+
+	// TODO handle not found
+
+	rec, err := doc.Get(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("Error while querying Firestore: %v", err)
+	}
+
+	var data structs.NotificationCounter
+	err = rec.DataTo(&data)
+	if err != nil {
+		return nil, fmt.Errorf("Error while querying Firestore: %v", err)
+	}
+	return &data, nil
+}
+
+func getActivationCounters(ctx context.Context, now time.Time) (yesterday int32, total int32, err error) {
+	logger := logging.FromContext(ctx)
+
+	var date = now.Add(time.Hour * -24).Format("20060102")
+
+	// yesterday
+
+	yesterdayData, err := getActivationCounter(ctx, date)
+	if err != nil {
+		return 0, 0, err
+	}
+
+	logger.Infof("Activations count for yesterday: %v", yesterdayData.UsersCount)
+
+	// total
+
+	totalData, err := getActivationCounter(ctx, "total")
+	if err != nil {
+		return 0, 0, err
+	}
+
+	logger.Infof("Activations total count: %v", totalData.UsersCount)
+
+	return int32(yesterdayData.UsersCount), int32(totalData.UsersCount), nil
+}
+
+func getActivationCounter(ctx context.Context, key string) (*structs.UserCounter, error) {
+	logger := logging.FromContext(ctx)
+	client := realtimedb.Client{}
+
+	logger.Debugf("Getting activation counter with key %v", key)
+
+	var data structs.UserCounter
+
+	// TODO handle not found
+
+	if err := client.NewRef(constants.DbUserCountersPrefix+key).Get(ctx, &data); err != nil {
+		logger.Debugf("Error while querying DB: %v", err)
+		return nil, err
+	}
+
+	return &data, nil
+}
+
+func getPublishersCount(ctx context.Context) (yesterdayCount int32, totalCount int32, err error) {
+	logger := logging.FromContext(ctx)
+
+	startOfTommorow := time.Now().UTC().Add(time.Hour * -24).Truncate(time.Hour * 24)
+	startOfErouska := time.Unix(1598918400, 0) // 1.9.2020
+
+	counts, err := getPublishersValues(ctx, startOfTommorow, 84600 /* 1 day */)
+	if err != nil {
+		logger.Debugf("Could not fetch data for publishers of last day: %v", err)
+		return 0, 0, err
+	}
+
+	yesterdayCount = counts[0] // get the newest from daily buckets
+
+	logger.Infof("Publishers count for yesterday %v", yesterdayCount)
+
+	counts, err = getPublishersValues(ctx, startOfErouska, 2592000 /* 1 month */)
+	if err != nil {
+		logger.Debugf("Could not fetch data for publishers of all time: %v", err)
+		return 0, 0, err
+	}
+
+	totalCount = sum(counts) // if there's multiple values, sum them all up - this is all time startOfErouska
+	totalCount -= publishersOffset
+
+	logger.Infof("Publishers total count: %v", totalCount)
+
+	return yesterdayCount, totalCount, nil
+}
+
+func getPublishersValues(ctx context.Context, from time.Time, sumWindow int64) ([]int32, error) {
+	monitoringClient := monitoring.Client{}
+
+	startOfToday := time.Now().UTC().Truncate(time.Hour * 24)
+
+	// this just adds the configuration... one would use function currying of Go supports such thing
+	return monitoringClient.ReadSummarized(ctx,
+		"steadfast-helix-287408", // TODO parametrize
```

Comment:
this cannot be shared with anyone!...

Reply:
I know. But how do you want to extract it? I made it env... but what its name should be? :roll_eyes:

Reply:
`METRICS_SOURCE_PROJECT`?

Reply:
ah I see it is METRICS_PROJECT_ID already, that is fine.